mods/sgssb/abilities.js

```diff
@@ -0,0 +1,20 @@
+"use strict";
+
+exports.BattleAbilities = {
+
+// Ashley the Pikachu
+	primalsurge: {
+		shortDesc: "On switch-in, this Pokemon gets a 2x Speed boost on Electric Terrain.",
+		onStart: function (source) {
+			this.addPseudoWeather('electricterrain', source);
```

Comment:
Im not sure if this is possible but if it is this ability is supposed to work like primal weather, lasting as long as the user is active.